libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/StackCallerProtectionDomainChainExtractor.java

```diff
@@ -24,6 +25,14 @@ public final class StackCallerProtectionDomainChainExtractor implements Function
      */
     public static final StackCallerProtectionDomainChainExtractor INSTANCE = new StackCallerProtectionDomainChainExtractor();
 
+    private static final StackWalker STACK_WALKER = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
+    private static final Set<String> ACCESS_CONTROLLER_CLASSES = Set.of(
```

Comment:
nit: I think we should add a comment here, why we'd need to support both classes.